src/constraints/outage_constraints.jl

```diff
@@ -385,3 +414,163 @@ function add_cannot_have_MG_with_only_PVwind_constraints(m, p)
+            )
+        end
+    end
+
+    # Fuel cell constraints
+    if !isempty(p.techs.fuel_cell)
+        if solver_is_compatible_with_indicator_constraints(p.s.settings.solver_name)
+            @constraint(m, [t in p.techs.fuel_cell, s in p.s.electric_utility.scenarios, tz in p.s.electric_utility.outage_start_time_steps, ts in p.s.electric_utility.outage_time_steps],
+                m[:binMGFCIsOnInTS][t] => { m[:dvMGRatedProduction][t, s, tz, ts] >= p.s.fuel_cell.min_turn_down_fraction * m[:dvMGsize][t] }
+            )
+            @constraint(m, [t in p.techs.fuel_cell, s in p.s.electric_utility.scenarios, tz in p.s.electric_utility.outage_start_time_steps, ts in p.s.electric_utility.outage_time_steps],
+                !m[:binMGFCIsOnInTS][t] => { m[:dvMGRatedProduction][t, s, tz, ts] <= 0 }
+            )
+        else
+            @constraint(m, [t in p.techs.fuel_cell, s in p.s.electric_utility.scenarios, tz in p.s.electric_utility.outage_start_time_steps, ts in p.s.electric_utility.outage_time_steps],
+                m[:dvMGRatedProduction][t, s, tz, ts] >= p.s.fuel_cell.min_turn_down_fraction * m[:dvMGsize][t] - p.max_sizes[t] * (1-m[:binMGFCIsOnInTS][s, tz, ts])
+            )
+            @constraint(m, [t in p.techs.fuel_cell, s in p.s.electric_utility.scenarios, tz in p.s.electric_utility.outage_start_time_steps, ts in p.s.electric_utility.outage_time_steps],
+                m[:dvMGRatedProduction][t, s, tz, ts] <= p.max_sizes[t] *  m[:binMGFCIsOnInTS][s, tz, ts]
+            )
+        end
+        @constraint(m, [t in p.techs.fuel_cell, s in p.s.electric_utility.scenarios, tz in p.s.electric_utility.outage_start_time_steps, ts in p.s.electric_utility.outage_time_steps],
+            m[:binMGTechUsed][t] >= m[:binMGFCIsOnInTS][s, tz, ts]
+        )
+    end
+
+    # Hydrogen storage constraints
+    # Initial SOC at start of each outage equals the grid-optimal SOC
+    @constraint(m, [b in p.s.storage.types.hydrogen, s in p.s.electric_utility.scenarios, tz in p.s.electric_utility.outage_start_time_steps],
+        m[:dvMGStoredEnergy][b, s, tz, 0] <= m[:dvStoredEnergy][b, tz]
+    )
+    # Min SOC
+    if p.s.storage.attr["HydrogenStorage"].soc_min_applies_during_outages
+        @constraint(m, [b in p.s.storage.types.hydrogen, s in p.s.electric_utility.scenarios, tz in p.s.electric_utility.outage_start_time_steps, ts in p.s.electric_utility.outage_time_steps],
+            m[:dvMGStoredEnergy][b, s, tz, ts] >=  p.s.storage.attr[b].soc_min_fraction * m[:dvStorageEnergy][b]
+        )
+    end
+    # State of charge upper bound is storage system size
+    @constraint(m, [b in p.s.storage.types.hydrogen, s in p.s.electric_utility.scenarios, tz in p.s.electric_utility.outage_start_time_steps, ts in p.s.electric_utility.outage_time_steps],
+        m[:dvStorageEnergy][b] >= m[:dvMGStoredEnergy][b, s, tz, ts]
+    )
+    
+    if p.s.electrolyzer.require_compression
```

Comment:
if statement unnecessary, could be one constraint that loops through t in union(p.techs.compressor, p.techs.electrolyzer) cause dvMGProductionToStorage should be fixed to zero for one or the other